### PR TITLE
Added support for Carbon 3.x

### PR DESCRIPTION
**Pull request by indy2kro**
- [ ] Added automated tests
- [ ] Documented for all relevant versions
- [ ] Updated the changelog

**Changes**

- Added support for Carbon 3.x
- Upgraded github actions to use node 20

**Breaking changes**

- Removed support for illuminate 8 and 9 since they are not compatible with Carbon 3.x

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,9 +17,9 @@
   "require": {
     "php": "^7.4 || ^8",
     "ext-calendar": "*",
-    "illuminate/support": "^8.73 || ^9 || ^10 || ^11",
```
[the remark this answers is not in the document]
Comment (indy2kro) [reply composer.json:20]: Well, they are not compatible with each other, I guess the combinations can be excluded just in GitHub actions though.

[the remark this answers is not in the document]
Comment (indy2kro) [reply composer.json:20]: Actually it seems that carbon is compatible as long as minimum version required is `^2.62.1`. Updated MR and all GitHub actions seem to be ok.

File: .github/workflows/validate.yml [hunk 2]
```diff
@@ -111,4 +111,4 @@ jobs:
 
       - run: vendor/bin/phpunit --coverage-clover=.build/logs/clover.xml
 
-      - uses: codecov/codecov-action@v3
+      - uses: codecov/codecov-action@v4
```
Comment (spawnia) [inline .github/workflows/validate.yml:114]:
```suggestion
      # Not using v4 due to the breaking changes described in https://github.com/codecov/codecov-action/releases/tag/v4.0.0
      - uses: codecov/codecov-action@v3
```

File: .gitignore [hunk 3]
```diff
@@ -2,3 +2,4 @@
 /.idea
 /vendor
 /composer.lock
+/.phpunit.result.cache
```
Comment (spawnia) [inline .gitignore:5]:
```suggestion
/.phpunit.result.cache

```

**Comment by spawnia** [comment]
Released with https://github.com/mll-lab/php-utils/releases/tag/v3.2.0

**Pull request merged**